packages/xpetra/CMakeLists.txt

```diff
@@ -149,12 +120,27 @@ IF (${PACKAGE_NAME}_ENABLE_Epetra)
           MESSAGE(FATAL_ERROR "If Xpetra Epetra support is enabled and Epetra 32-bit global indices are enabled, Xpetra requires that you enable Tpetra_INST_INT_INT")
         ELSE()
           # Epetra was enabled but only implicitly; give a warning but turn off Xpetra's Epetra support
-          MESSAGE(WARNING "32-bit global indices are enabled in Epetra but not Tpetra. Since Epetra was not enabled explicitly by user, disabling Epetra support in Xpetra.")
-          SET(${PACKAGE_NAME}_ENABLE_Epetra OFF)
+          SET(WARN_MSG "32-bit global indices are enabled in Epetra but not Tpetra. Since Epetra was not enabled explicitly by user, **DISABLING EPETRA SUPPORT** in Xpetra and: ")
+          DUAL_SCOPE_SET(${PACKAGE_NAME}_ENABLE_Epetra OFF)
+          DUAL_SCOPE_SET(${PACKAGE_NAME}_ENABLE_EpetraExt OFF)
+          DUAL_SCOPE_SET(${PACKAGE_NAME}_Epetra_NO_32BIT_GLOBAL_INDICES ON)
+          GLOBAL_SET(${PACKAGE_NAME_UC}_EPETRA_NO_32BIT_GLOBAL_INDICES ON)
+          IF(${PROJECT_NAME}_ENABLE_Zoltan2)
+            SET(WARN_MSG "${WARN_MSG} Zoltan2")
+            SET(Zoltan2_ENABLE_Epetra OFF PARENT_SCOPE)
+          ENDIF()
+          IF(${PROJECT_NAME}_ENABLE_MueLu)
+            SET(WARN_MSG "${WARN_MSG} MueLu")
+            SET(MueLu_ENABLE_Epetra OFF PARENT_SCOPE)
```

Comment:
Likewise, you might want to print "NOTE: Setting MueLu_ENABLE_Epetra=OFF because Xpetra_ENBALE_Epetra=OFF" .